keras_preprocessing/image/iterator.py

```diff
@@ -82,6 +82,7 @@ def _flow_index(self):
             if self.batch_index == 0:
                 self._set_index_array()
 
+            assert self.n != 0, "Won't iterate over 0 number of images"
```

Comment:
The best would be to raise an Exception.

Reply:
- Moved this to __init__
- Added `FileNotFoundError`

Reply:
- Moved the same to `_flow_index` method of `Iterator` class
- Raising `Exception` since CI tool gave `Unknown: FileNotFoundError`

Reply:
instead of raising on exception it should just iterate returning zero sample on each iteration. This can be accomplish by something like:
```
            if self.n == 0:
                current_index = 0
            else:
                current_index = (self.batch_index * self.batch_size) % self.n 
```

Reply:
Agreed over the suggestion.